src/js/angular/security/controllers.js

```diff
@@ -104,6 +104,37 @@ const parseAuthorities = function (authorities) {
     };
 };
 
+const compactAuthorities = function (authorities) {
+    const writeAllRepos = authorities.find(authority => authority === `${WRITE_REPO_PREFIX}*`);
+    if (writeAllRepos) {
+        return  _.remove(authorities, function (_authority) {
```

Comment:
You know that the function `_.remove` from lodash actually mutates the array argument, aren't you? Is it intentional that you just use the return value of the function and not aware of the actual authorities array? Isn't it better to use Array.reduce or Array.filter for example instead?

Reply:
The idea is to modify the array, leaving only "WRITE_REPO_*", "READ_REPO_*" and all other authorities, that aren't related with read or write ones.